src/derive.rs

```diff
@@ -363,7 +415,57 @@ impl<'a> DerivedModule<'a> {
                             naga::RayQueryFunction::Terminate => naga::RayQueryFunction::Terminate,
                         },
                     },
-
+                    Statement::SubgroupBallot { result, predicate } => Statement::SubgroupBallot {
+                        result: map_expr!(result),
+                        predicate: map_expr_opt!(predicate),
+                    },
+                    Statement::SubgroupGather {
+                        mode,
+                        argument,
+                        result,
+                    } => match mode {
+                        GatherMode::BroadcastFirst => Statement::SubgroupGather {
+                            mode: *mode,
+                            argument: map_expr!(argument),
+                            result: map_expr!(result),
+                        },
+                        GatherMode::Broadcast(hnd) => Statement::SubgroupGather {
+                            mode: GatherMode::Broadcast(map_expr!(hnd)),
+                            argument: map_expr!(argument),
+                            result: map_expr!(result),
+                        },
+                        GatherMode::Shuffle(hnd) => Statement::SubgroupGather {
+                            mode: GatherMode::Shuffle(map_expr!(hnd)),
+                            argument: map_expr!(argument),
+                            result: map_expr!(result),
+                        },
+                        GatherMode::ShuffleDown(hnd) => Statement::SubgroupGather {
+                            mode: GatherMode::ShuffleDown(map_expr!(hnd)),
+                            argument: map_expr!(argument),
+                            result: map_expr!(result),
+                        },
+                        GatherMode::ShuffleUp(hnd) => Statement::SubgroupGather {
+                            mode: GatherMode::ShuffleUp(map_expr!(hnd)),
+                            argument: map_expr!(argument),
+                            result: map_expr!(result),
+                        },
+                        GatherMode::ShuffleXor(hnd) => Statement::SubgroupGather {
+                            mode: GatherMode::ShuffleXor(map_expr!(hnd)),
+                            argument: map_expr!(argument),
+                            result: map_expr!(result),
+                        },
+                    },
```

Comment:
```suggestion
                    } => {
                        let mut mode = mode.clone();
                        match mode {
                            GatherMode::BroadcastFirst => (),
                            GatherMode::Broadcast(ref mut h_src)
                            | GatherMode::Shuffle(ref mut h_src)
                            | GatherMode::ShuffleDown(ref mut h_src)
                            | GatherMode::ShuffleUp(ref mut h_src)
                            | GatherMode::ShuffleXor(ref mut h_src) => *h_src = map_expr!(h_src),
                        };
                        Statement::SubgroupGather {
                            mode,
                            argument: map_expr!(argument),
                            result: map_expr!(result),
                        }
                    }
```
sorry i know this is picky